sdk/go/client.go

```diff
@@ -37,24 +37,17 @@ func WithWorkdir(path string) ClientOpt {
 	})
 }
 
-// WithConfigPath sets the engine config path
-func WithConfigPath(path string) ClientOpt {
-	return clientOptFunc(func(cfg *engineconn.Config) {
-		cfg.ConfigPath = path
-	})
-}
-
-// WithNoExtensions disables installing extensions
-func WithNoExtensions() ClientOpt {
+// WithLogOutput sets the progress writer
+func WithLogOutput(writer io.Writer) ClientOpt {
 	return clientOptFunc(func(cfg *engineconn.Config) {
-		cfg.NoExtensions = true
+		cfg.LogOutput = writer
 	})
 }
 
-// WithLogOutput sets the progress writer
-func WithLogOutput(writer io.Writer) ClientOpt {
+// WithConn sets the engine connection explicitly
+func WithConn(conn engineconn.EngineConn) ClientOpt {
```

Comment:
Want to make sure this doesn't get lost in the PR, this enables the go sdk be set with a custom EngineConn. I wanted this because `dagger do` has to make a lot of API calls to the Project API and writing them in raw graphql was driving me insane (besides not really being maintainable since changes to the api will require manual changes to graphql query strings).

This change enables it to create a go sdk client directly from the `router.Router` object it receives when it starts the engine.

This is technically a feature which is very special to the Go SDK, couldn't have an equivalent in others, but I can't really think of why that's actually a big deal in this particular case.

I also don't think this breaks any of the ideas around enabling default clients that aren't created w/ an explicit `.Connect`. When we do that, we'll still need some mechanism to create clients manually for a bunch of different reasons, this can just be part of that.

Just calling it out because I have a vague memory of there being hesitation around this feature in the past (could be misremembering), so want to make sure we're all on the same page (cc @aluzzardi)

Reply:
Neat!